gogs_client/_implementation/http_utils.py

```diff
@@ -11,8 +11,9 @@ class RelativeHttpRequestor(object):
     A thin wrapper around the requests module that allows for endpoint paths
     to be given relative to a fixed base URL
     """
-    def __init__(self, base_url):
+    def __init__(self, base_url, session=requests.Session()):
```

Comment:
I think it would be safer to have `session=None`, and then check if `session is None` inside the constructor. If for some reason I created two `GogsApi` obects (each using the default session), they would be sharing the same `request.Session()` object.

In general, having mutable default arguments is risky business in Python.